internal/archive/tar/tar.go

```diff
@@ -0,0 +1,76 @@
+package tar
+
+import (
+	"archive/tar"
+	"bytes"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/saucelabs/saucectl/internal/sauceignore"
+)
+
+// Archive archives the resource and exclude files and folders based on sauceignore logic.
+func Archive(src string, matcher sauceignore.Matcher) (io.Reader, error) {
+	bb := new(bytes.Buffer)
+	w := tar.NewWriter(bb)
+	defer w.Close()
+
+	walker := func(file string, fileInfo os.FileInfo, err error) error {
+		// Only will be applied if we have .sauceignore file and have patterns to exclude files and folders
+		if matcher.Match(strings.Split(file, string(os.PathSeparator)), fileInfo.IsDir()) {
+			return nil
+		}
+
+		header, err := tar.FileInfoHeader(fileInfo, file)
+		if err != nil {
+			return err
+		}
+
+		relFilePath := file
+		if filepath.IsAbs(src) {
+			/*
+				for temporary files filepath.Rel would return "." as a relative path,
+				that's why we need to make an exception for temporary files and return only the last element
+			*/
+			if strings.Contains(src, os.TempDir()) {
```

Comment:
As discussed via chat, this ought to be addressed before we proceed further.
To re-iterate:
- `archive.TarResource()` didn't need any special handling for temp files, neither should this
- all in all, `zip` and `tar` package should behave the same way and package the same way and only differ in the actual archive that's put out

Reply:
I refactor the code and now the output is the same as for the `zip` package. Please have a look one more time.